### PR TITLE
Add the initial version of CI using GitHub Actions

### PR DESCRIPTION
**Pull request by yTakatsukasa**
This is simple but limited version of CI on GitHub Actions.
Though this is not fully featured, but I believe this is better than nothing.

I think GitHub actions is faster than travis especially downloading packages and restoring ccache.
The entire action takes about 12 min. with ccache while the initial run that is without ccache takes 45 min.

#### supported
- gcc with ccache (cache will be reused) on Ubuntu-18.04
- vlt and vltmt including vcddiff

#### not yet (future work, or on Google cloud)
- clang
- Other platforms (Ubuntu-16.04, Ubuntu-20.04, mac OS)
- SystemC
- code coverage

### CODE REVIEW

**Comment by yTakatsukasa** [comment]
[This](https://github.com/yTakatsukasa/verilator/runs/1498546893?check_suite_focus=true) is the result of this .yml executed on my repository.

File: .github/workflows/check_pr.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: Verilator PR ci
```
Comment (wsnyder) [inline .github/workflows/check_pr.yml:1]:
```suggestion
name: Verilator PR CI
```

Sorry the mixed case bothers me ;)

File: .github/workflows/check_pr.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+              with:
+                  path: ${{ github.workspace }}/.ccache
+                  key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ github.ref }}-${{ github.sha }}
+                  restore-keys: |
+                      ${{ runner.os }}-build-${{ env.cache-name }}-
+                      ${{ runner.os }}-build-
+                      ${{ runner.os }}-
+            - name: Build
+              env:
+                  TRAVIS_BUILD_STAGE_NAME: build
```
Comment (wsnyder) [inline .github/workflows/check_pr.yml:31]:
```suggestion
                  CI_BUILD_STAGE_NAME: build
```

Can you please include converting the variable names to CI_ so these will be less confusing and more general.

**Review by wsnyder — COMMENTED** [review]
Great!  Runs look good too.

Can you please remove old travis files, as we won't want to support them.

Please use 20.04, as that has libsystemc libsystemc-dev so we can run SystemC.  Also it's what I use so want to match versions.

Please update docs/internals.adoc which mentions Travis.

Appreciated!

**Comment by yTakatsukasa** [comment]
Travis specific keywords are gone.
Now they have `CI_` prefix.

OS is Ubuntu-20.04 and SystemC is installed.
Now the number of skipped tests are much less.

**Comment by wsnyder** [comment]
If it looks like it is living feel free to push to master.  Would love additional clang, etc, if you have time, but at least we then are back into CI.

**Comment by yTakatsukasa** [comment]
Sure. Now I can send other PRs for new features.

CI looks triggerred as expected. 
https://github.com/verilator/verilator/runs/1499629432?check_suite_focus=true

**Comment by wsnyder** [comment]
Note I renamed the workflow so badges would work (spaces caused a problem).

**Comment by yTakatsukasa** [comment]
Thank you for polish.

I will send a PR that uses both gcc and clang soon.

**Pull request merged**